source/Hud.hx

```diff
@@ -62,12 +75,12 @@ class Hud extends FlxTypedGroup<FlxSprite> {
 
     public function incrementLives()
     {
-        livesCounter.text = "" + (++lives);
+        livesCounter.text = "" + (++player.health);
     }
 
     public function decrementLives()
     {
-        livesCounter.text = "" + (--lives);
+        livesCounter.text = "" + (--player.health);
```

Comment:
I don't think we want to have the HUD class responsible for updating the value of the player's health, the real responsibility of this class is displaying the state of things as they are. Saying something like
```
public updatelivesDisplay()
{
    livesCounter.text = "" + player.health;
}
``` 
is good. The only tricky thing is making sure it's called at an appropriate time. 

You could throw a call to this inside this class's `update()` method, you could try changing the `Player` class to make it call this method when health changes, or you could do even trickier things to really keep these classes decoupled.

Now that Sam's branch has been merged to master, you can throw code into the player class to take advantage of what he set up too, which might be helpful.